test/integration/Test/Integration/Scenario/Wallets.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TypeApplications #-}
+
+
+module Test.Integration.Scenario.Wallets
+    ( spec
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Api.Types
+    ( ApiWallet )
+import Cardano.Wallet.Primitive.Types
+    ( WalletBalance (..), WalletDelegation (..), WalletState (..) )
+import Data.Quantity
+    ( Quantity (..), zeroPercentage )
+import qualified Network.HTTP.Types.Status as HTTP
+import Test.Hspec
+    ( SpecWith, it )
+import Test.Integration.Framework.DSL
+    ( Context (..)
+    , Headers (..)
+    , Payload (..)
+    , addressPoolGap
+    , balance
+    , balanceAvailable
+    , balanceTotal
+    , delegation
+    , expectFieldEqual
+    , expectResponseCode
+    , getFromResponse
+    , json
+    , passphraseLastUpdate
+    , request
+    , state
+    , walletId
+    , walletName
+    )
+
+spec :: SpecWith Context
+spec = do
+    it "Create a wallet" $ \ctx -> do
+
+        let payload = Json [json| {
+                "name": "1st Wallet",
+                "mnemonic_sentence": ["identify", "screen", "lock", "bargain", "inch", "drop", "canyon", "flock", "dry", "zone", "wash", "argue", "system", "glory", "light"],
+                "mnemonic_second_factor": ["attract", "tornado", "slender", "pumpkin", "clown", "announce", "term", "winner", "ready"],
+                "passphrase": "Secure Passphrase",
+                "address_pool_gap": 20
+                } |]
+        r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+        print (show r)
+        expectResponseCode @IO HTTP.status200 r
+        expectFieldEqual walletName "1st Wallet" r
+        expectFieldEqual addressPoolGap 20 r
+        expectFieldEqual balance (WalletBalance
+                                    { available = Quantity 0
+                                    , total = Quantity 0}) r
```

Comment:
I'd recommend to use an intermediate var here to avoid having to indent things like this :/ Or, drop the field constructors: `WalletBalance (Quantity 0) (Quantity 0)`